server/src/main/java/org/opensearch/rest/pagination/IndexBasedPaginationStrategy.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.pagination;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.common.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This strategy can be used by the Rest APIs wanting to paginate the responses based on Indices.
+ * The strategy considers create timestamps of indices as the keys to iterate over pages.
+ *
+ * @opensearch.internal
+ */
+public class IndexBasedPaginationStrategy implements PaginationStrategy<String> {
+
+    private final PaginatedQueryResponse paginatedQueryResponse;
+    private final List<String> indicesFromRequestedToken;
+
+    public IndexBasedPaginationStrategy(PaginatedQueryRequest paginatedQueryRequest, String paginatedElement, ClusterState clusterState) {
+        IndexStrategyTokenParser requestedToken = paginatedQueryRequest.getRequestedTokenStr() == null
+            ? null
+            : new IndexStrategyTokenParser(paginatedQueryRequest.getRequestedTokenStr());
+
+        // Get sorted list of indices not containing the ones created after query start time
+        List<String> sortedIndicesList = PaginationStrategy.getListOfIndicesSortedByCreateTime(
+            clusterState,
+            paginatedQueryRequest.getSort(),
+            requestedToken == null ? Long.MAX_VALUE : requestedToken.queryStartTime
+        );
+        if (sortedIndicesList.isEmpty()) {
+            // Denotes, that all the indices which were created before the queryStartTime have been deleted.
```

Comment:
why is this required ? Can't we return the indices even if they are created after queryStartTime ?